demo/Makefile.inc

```diff
@@ -0,0 +1,2 @@
+# Flags
+CFLAGS += -std=c99 -pedantic -Wall -Wextra -O2
```

Comment:
Wouldn't it be simpler to just add this as recommendation/requirement to the [Reviewers Guide](https://github.com/Immediate-Mode-UI/Nuklear#reviewers-guide ) instead of complicating the build process and introducing a new file for such a simple purpose?

Reply:
Is it wanted to print those warnings? If so, I can also include the flags in every makefile - no problem. But I would first like to discuss the flags.

Reply:
Generally the stricter the better. Though we do **not require** such flags for non-`nuklear.h` stuff as stated in the Reviewers Guide. It's just a preference. If it makes some `demo/`s or `example/`s issue a warning, then these should be fixed together with introduction of these flags.

Regarding each flag separately, I think some demos do not comply with C99, but maybe with newer version or some dialect (GNU extensions etc.) which means it might really need some rework.